engine/runtime/src/main/java/org/enso/interpreter/caches/SuggestionsCache.java

```diff
@@ -146,56 +136,13 @@ public String blobHash(Metadata meta) {
     return meta.blobHash();
   }
 
-  // Suggestions class is not a record because of a Frgaal bug leading to invalid compilation error.
-  public static final class Suggestions implements Serializable {
-
-    private final List<Suggestion> suggestions;
-
-    public Suggestions(List<Suggestion> suggestions) {
-      this.suggestions = suggestions;
-    }
-
-    public List<Suggestion> getSuggestions() {
-      return suggestions;
-    }
-  }
-
-  // CachedSuggestions class is not a record because of a Frgaal bug leading to invalid compilation
-  // error.
-  public static final class CachedSuggestions {
-
-    private final LibraryName libraryName;
-    private final Suggestions suggestions;
-
-    private final Optional<List<SourceFile<TruffleFile>>> sources;
-
-    public CachedSuggestions(
-        LibraryName libraryName,
-        Suggestions suggestions,
-        Optional<List<SourceFile<TruffleFile>>> sources) {
-      this.libraryName = libraryName;
-      this.suggestions = suggestions;
-      this.sources = sources;
-    }
-
-    public LibraryName getLibraryName() {
-      return libraryName;
-    }
-
-    public Optional<List<SourceFile<TruffleFile>>> getSources() {
-      return sources;
-    }
-
-    public Suggestions getSuggestionsObjectToSerialize() {
-      return suggestions;
-    }
-
-    public List<Suggestion> getSuggestions() {
-      return suggestions.getSuggestions();
-    }
-  }
+  /**
+   * @param libraryName
+   * @param suggestions Must not be null.
+   */
+  public record CachedSuggestions(LibraryName libraryName, ArrayList<Suggestion> suggestions) {}
```

Comment:
Shouldn't we prefer `java.util.List` over the implementation type?

Reply:
I have tried that with `java.util.List`, but it does not work, since it is an interface, and `persistance-dsl` does not accept interfaces - it fails with an error message like: "java.util.List must contain one public parameterless constructor".